dot/state/initialize.go

```diff
@@ -149,7 +149,7 @@ func loadGrandpaAuthorities(t *trie.Trie) ([]types.GrandpaVoter, error) {
 
 	r := &bytes.Buffer{}
 	_, _ = r.Write(authsRaw[1:])
-	return types.DecodeGrandpaVoters(r)
+	return types.DecodeGrandpaVoters(authsRaw[1:])
```

Comment:
as `DecodeGrandpaVoters` receive `authsRaw[1:]` directly so the `&bytes.Buffer` is not needed anymore and you can remove it